pkg/config/deployment.go

```diff
@@ -59,18 +59,43 @@ type DeploymentPlanner struct {
 }
 
 type Trigger struct {
-	// Configuration in case changes on commit are found.
+	// Configurable fields used while deciding the application
+	// should be triggered or not based on commit changes.
 	OnCommit OnCommit `json:"onCommit"`
+	// Configurable fields used while deciding the application
+	// should be triggered or not based on received SYNC command.
+	OnCommand OnCommand `json:"onCommand"`
+	// Configurable fields used while deciding the application
+	// should be triggered or not based on OUT_OF_SYNC state.
+	OnOutOfSync OnOutOfSync `json:"onOutOfSync"`
 }
 
 type OnCommit struct {
-	// Control trigger new deployment on Git change or not.
+	// Whether to exclude application from triggering target
+	// when a new commit touched the application.
+	// Default is false.
 	Disabled bool `json:"disabled,omitempty"`
 	// List of directories or files where their changes will trigger the deployment.
 	// Regular expression can be used.
 	Paths []string `json:"paths,omitempty"`
 }
 
+type OnCommand struct {
+	// Whether to exclude application from triggering target
+	// when received a new SYNC command.
+	// Default is false.
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+type OnOutOfSync struct {
+	// Whether to exclude application from triggering target
+	// when application is at OUT_OF_SYNC state.
+	// Default is true.
+	Disabled bool `json:"disabled,omitempty" default:"true"`
```

Comment:
nits, with this `default:"true"` labels on boolean type value, even if users explicitly set `spec.input.trigger.onOutOfSync.Disable: false`, they still can not enable trigger new deployment on out of sync.

Reply:
Sorry, what do you mean?
Why " they still can not enable trigger new deployment on out of sync"?

Reply:
I think I got your point. The "defaults" lib is configuring that field to be `true` whenever its value is `false`, doesn't matter if it was specified explicitly or not. So its final value always will be `true`, right?

Reply:
Yes, so I think we may need `*bool` for this case 🤔

Reply:
Right. *bool or defined a new Bool type for that case. Let me fix it. Nice catch!

Reply:
btw, other places (such as `autoRollback` configuration) have this issue as well, so addressing that issue by a separated PR is fine, but I just want to explicitly mention it to ensure everyone aware of that issue.

Reply:
> Right. *bool or defined a new Bool type for that case. Let me fix it. Nice catch!

Thanks, will fix other places by separated PR later 👍

Reply:
@khanhtc1202 I have added a TODO to this PR. And let me fix all of those things by a separate PR.

Reply:
🙆‍♀️